src/harness.ts

```diff
@@ -47,6 +47,28 @@ const EVENT_HANDLERS = [
 ];
 
 /**
+ * An internal function which finds a VNode based on `key`
+ * @param target The VNode to search
+ * @param key The key to find
+ */
+function findVNodebyKey(target: VNode, key: string | object): VNode | undefined {
```

Comment:
I would ask, what if there are multiple nodes with the same key? This is actually possible, as the keys are only needed to differentiate between siblings and not parent/children?